ratelimits/src/index.ts

```diff
@@ -0,0 +1,71 @@
+import Limits from './limits.json'
+const limits: Record<string, any> = Limits // ugly solution to dynamically reference JSON properties
+
+const BURST_UNDEFINED_QUOTA_MULTIPLE = 2
+
+export interface ProviderRateLimit {
+  burst: number
+  quota: number
+  second: number
+  minute: number
+}
+
+export type DeclaredTier = {
+  rateLimit1s?: number
+  rateLimit1m?: number
+  rateLimit1h?: number
+  tierName: string
+}
+
+const calculateLimits = (declaredTier: DeclaredTier) => {
+  let quota: number
+  if (declaredTier.rateLimit1h) {
+    quota = declaredTier.rateLimit1h / 60 // quota is based on per-minute
+  } else if (declaredTier.rateLimit1m) {
+    quota = declaredTier.rateLimit1m
+  } else if (declaredTier.rateLimit1s) {
+    quota = declaredTier.rateLimit1s * 60
+  } else {
+    throw Error('at least one of the rateLimits must be defined')
+  }
+
+  let burst: number
+  if (declaredTier.rateLimit1s) {
+    burst = declaredTier.rateLimit1s
+  } else {
+    burst = (quota / 60) * BURST_UNDEFINED_QUOTA_MULTIPLE
+  }
+
+  return [burst, quota]
+}
+
+const findTier = (provider: string, tier?: number, tierName?: string): DeclaredTier => {
+  const matchedTier =
+    tier !== undefined
+      ? limits[provider][tier]
+      : limits[provider].find((element: Record<string, unknown>) => element.tierName === tierName)
+  if (matchedTier === undefined) {
+    throw Error("tier or tierName doesn't match provider spec in limits.json")
+  }
+  return matchedTier
+}
+
+export const getRateLimit = (
```

Comment:
Might be better DevEx to take a required `tier` and determine if it parses to an Int or is a string (name)